### DISCUSSION TITLE
can workflow-controller connect to apiserver by kubeconfig

### DISCUSSION THREAD

**Discussion by chaosi-zju** [argoproj/argo-workflows#12012] [Q&A] [answered]
As we known, workflow-controller connect to apiserver by serveracount by default, I wanna how can I deploy workflow-controller so that it connect to apiserver by specified kubeconfig?

**Comment by agilgur5** [comment, marked as answer]
Yes it can. The `workflow-controller-configmap` has a `kubeConfig` field for this purpose.
See [these lines in the example ConfigMap](https://github.com/argoproj/argo-workflows/blob/f11c02ace54244c2e0f38908a80a10bc07bf8fc4/docs/workflow-controller-configmap.yaml#L44-L55) for how to use the `kubeConfig` field.
